### ISSUE TITLE
Use system standard config directory

### ISSUE THREAD

**Issue by gdestuynder** [mozilla-iam/mozilla-aws-cli#133]
Normally I'd find the config in `~/.config` according to XDG, but in this case it's at `~/.maws/config`.
Not a huge issue, but most programs now try to be good citizens and store in `~/.config/APPNAME` and so should we :)

See also https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Note: this also works on OSX, Windows

**Comment by april** [comment]
The one problem with this is that the program also has a `credentials` file and a creates a local `cache` directory. Where should those go?

**Comment by gdestuynder** [comment]
cache goes in `$XDG_CACHE_HOME/APPNAME` and config goes in `$XDG_CONFIG_HOME/APPNAME` and these usually default to `~/.cache/APPNAME` and `~/.config/APPNAME` respectively. I found https://wiki.archlinux.org/index.php/XDG_Base_Directory to be a nice breakdown of the spec.

There's a couple of python libs that are nice to address this: https://pypi.org/project/appdirs/ and https://pypi.org/project/xdgappdirs/

The former follows platform conventions (between macOS, Linux and Windows) and the later does the same, except if your environment specifies XDG variables, in which case it always follows XDG (Linux). The advantage of that is for people who run mixed macOS/Linux or Windows/Linux userland, but it's more of a preference than a standard. I think either is totally acceptable though.

**Comment by april** [comment]
I'm pretty sure we discussed this, but where should I put the `credentials` file under this system, especially since it is not meant to be readable? At least on my system, `~/.cache` is `775`.

**Comment by gdestuynder** [comment]
i think it'd create the file `0600`, usually its like this (in shell, of course you'd want this with real code):
```
mkdir $XDG_CACHE_DIR/.maws/
chown 0750 $XDG_CACHE_DIR/.maws/
```

then create with mode `0600` such as:
```
import os

with open(os.open('XDG_CACHE_DIR_THINGS/.maws/credentials', os.O_CREAT | os.O_WRONLY, 0o600), 'w') as fd:
    fd.write('stuff')
```

**Comment by april** [comment]
Would you put the config as `~/.config/.maws/config` or `~/.config/maws`?

Sorry for all the questions, I'm just not super familiar with this.  :)

**Comment by gdestuynder** [comment]
I would use `~/.config/.maws/config` (or in fact `~/.config/.maws/settings.ini` since its an ini file) mainly because if you ever need to add more configs there in the future, that way you dont need up with both `~/.config/.maws/...` and `~/.config/.maws.ini` which would be annoying/confusing for users

Note I would definitely use something like https://pypi.org/project/appdirs/ and not hardcode the path or the lookup of the XDG environment variables so that everything works cross-platform

**Comment by gene1wood** [comment]
This is fixed in #178
